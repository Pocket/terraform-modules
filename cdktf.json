{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
