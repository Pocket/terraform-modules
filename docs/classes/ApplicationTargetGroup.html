<!DOCTYPE html><html class="default"><head><meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="IE=edge"/><title>ApplicationTargetGroup | @pocket/terraform-modules</title><meta name="description" content="Documentation for @pocket/terraform-modules"/><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="stylesheet" href="../assets/style.css"/><link rel="stylesheet" href="../assets/highlight.css"/><script async src="../assets/search.js" id="search-script"></script></head><body><script>document.body.classList.add(localStorage.getItem("tsd-theme") || "os")</script><header><div class="tsd-page-toolbar"><div class="container"><div class="table-wrap"><div class="table-cell" id="tsd-search" data-base=".."><div class="field"><label for="tsd-search-field" class="tsd-widget search no-caption">Search</label><input type="text" id="tsd-search-field"/></div><ul class="results"><li class="state loading">Preparing search index...</li><li class="state failure">The search index is not available</li></ul><a href="../index.html" class="title">@pocket/terraform-modules</a></div><div class="table-cell" id="tsd-widgets"><div id="tsd-filter"><a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a><div class="tsd-filter-group"><div class="tsd-select" id="tsd-filter-visibility"><span class="tsd-select-label">All</span><ul class="tsd-select-list"><li data-value="public">Public</li><li data-value="protected">Public/Protected</li><li data-value="private" class="selected">All</li></ul></div> <input type="checkbox" id="tsd-filter-inherited" checked/><label class="tsd-widget" for="tsd-filter-inherited">Inherited</label><input type="checkbox" id="tsd-filter-externals" checked/><label class="tsd-widget" for="tsd-filter-externals">Externals</label></div></div><a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a></div></div></div></div><div class="tsd-page-title"><div class="container"><ul class="tsd-breadcrumb"><li><a href="../modules.html">@pocket/terraform-modules</a></li><li><a href="ApplicationTargetGroup.html">ApplicationTargetGroup</a></li></ul><h1>Class ApplicationTargetGroup </h1></div></div></header><div class="container container-main"><div class="row"><div class="col-8 col-content"><section class="tsd-panel tsd-hierarchy"><h3>Hierarchy</h3><ul class="tsd-hierarchy"><li><span class="tsd-signature-type">Resource</span><ul class="tsd-hierarchy"><li><span class="target">ApplicationTargetGroup</span></li></ul></li></ul></section><section class="tsd-panel-group tsd-index-group"><h2>Index</h2><section class="tsd-panel tsd-index-panel"><div class="tsd-index-content"><section class="tsd-index-section "><h3>Constructors</h3><ul class="tsd-index-list"><li class="tsd-kind-constructor tsd-parent-kind-class tsd-is-overwrite"><a href="ApplicationTargetGroup.html#constructor" class="tsd-kind-icon">constructor</a></li></ul></section><section class="tsd-index-section "><h3>Properties</h3><ul class="tsd-index-list"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a href="ApplicationTargetGroup.html#node" class="tsd-kind-icon">node</a></li><li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a href="ApplicationTargetGroup.html#stack" class="tsd-kind-icon">stack</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="ApplicationTargetGroup.html#targetGroup" class="tsd-kind-icon">target<wbr/>Group</a></li></ul></section><section class="tsd-index-section tsd-is-inherited tsd-is-external"><h3>Methods</h3><ul class="tsd-index-list"><li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a href="ApplicationTargetGroup.html#toString" class="tsd-kind-icon">to<wbr/>String</a></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited tsd-is-static tsd-is-external"><a href="ApplicationTargetGroup.html#isConstruct" class="tsd-kind-icon">is<wbr/>Construct</a></li></ul></section></div></section></section><section class="tsd-panel-group tsd-member-group "><h2>Constructors</h2><section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class tsd-is-overwrite"><a id="constructor" class="tsd-anchor"></a><h3 class="tsd-anchor-link">constructor<a href="#constructor" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class tsd-is-overwrite"><li class="tsd-signature tsd-kind-icon">new <wbr/>Application<wbr/>Target<wbr/>Group<span class="tsd-signature-symbol">(</span>scope<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Construct</span>, name<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span>, config<span class="tsd-signature-symbol">: </span><a href="../interfaces/ApplicationTargetGroupProps.html" class="tsd-signature-type" data-tsd-kind="Interface">ApplicationTargetGroupProps</a><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="ApplicationTargetGroup.html" class="tsd-signature-type" data-tsd-kind="Class">ApplicationTargetGroup</a></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><p>Overrides Resource.constructor</p><ul><li>Defined in <a href="https://github.com/Pocket/terraform-modules/blob/51606c6/src/base/ApplicationTargetGroup.ts#L15">src/base/ApplicationTargetGroup.ts:15</a></li></ul></aside><h4 class="tsd-parameters-title">Parameters</h4><ul class="tsd-parameters"><li><h5>scope: <span class="tsd-signature-type">Construct</span></h5></li><li><h5>name: <span class="tsd-signature-type">string</span></h5></li><li><h5>config: <a href="../interfaces/ApplicationTargetGroupProps.html" class="tsd-signature-type" data-tsd-kind="Interface">ApplicationTargetGroupProps</a></h5></li></ul><h4 class="tsd-returns-title">Returns <a href="ApplicationTargetGroup.html" class="tsd-signature-type" data-tsd-kind="Class">ApplicationTargetGroup</a></h4></li></ul></section></section><section class="tsd-panel-group tsd-member-group "><h2>Properties</h2><section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a id="node" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagReadonly">Readonly</span> node<a href="#node" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">node<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">Node</span></div><aside class="tsd-sources"><p>Inherited from Resource.node</p><ul><li>Defined in node_modules/constructs/lib/construct.d.ts:247</li></ul></aside><div class="tsd-comment tsd-typography"><div class="lead">
<p>The tree node.</p>
</div></div></section><section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a id="stack" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagReadonly">Readonly</span> stack<a href="#stack" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">stack<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">TerraformStack</span></div><aside class="tsd-sources"><p>Inherited from Resource.stack</p><ul><li>Defined in node_modules/cdktf/lib/resource.d.ts:16</li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class"><a id="targetGroup" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagReadonly">Readonly</span> target<wbr/>Group<a href="#targetGroup" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">target<wbr/>Group<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">AlbTargetGroup</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/Pocket/terraform-modules/blob/51606c6/src/base/ApplicationTargetGroup.ts#L13">src/base/ApplicationTargetGroup.ts:13</a></li></ul></aside></section></section><section class="tsd-panel-group tsd-member-group tsd-is-inherited tsd-is-external"><h2>Methods</h2><section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a id="toString" class="tsd-anchor"></a><h3 class="tsd-anchor-link">to<wbr/>String<a href="#toString" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class tsd-is-inherited tsd-is-external"><li class="tsd-signature tsd-kind-icon">to<wbr/>String<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><p>Inherited from Resource.toString</p><ul><li>Defined in node_modules/constructs/lib/construct.d.ts:261</li></ul></aside><div class="tsd-comment tsd-typography"><div class="lead">
<p>Returns a string representation of this construct.</p>
</div></div><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">string</span></h4></li></ul></section><section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class tsd-is-inherited tsd-is-static tsd-is-external"><a id="isConstruct" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagStatic">Static</span> is<wbr/>Construct<a href="#isConstruct" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class tsd-is-inherited tsd-is-static tsd-is-external"><li class="tsd-signature tsd-kind-icon">is<wbr/>Construct<span class="tsd-signature-symbol">(</span>x<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">x</span><span class="tsd-signature-symbol"> is </span><span class="tsd-signature-type">Construct</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><p>Inherited from Resource.isConstruct</p><ul><li>Defined in node_modules/constructs/lib/construct.d.ts:243</li></ul></aside><div class="tsd-comment tsd-typography"><div class="lead">
<p>Checks if <code>x</code> is a construct.</p>
</div><div><p>Use this method instead of <code>instanceof</code> to properly detect <code>Construct</code>
instances, even when the construct library is symlinked.</p>
<p>Explanation: in JavaScript, multiple copies of the <code>constructs</code> library on
disk are seen as independent, completely different libraries. As a
consequence, the class <code>Construct</code> in each copy of the <code>constructs</code> library
is seen as a different class, and an instance of one class will not test as
<code>instanceof</code> the other class. <code>npm install</code> will not create installations
like this, but users may manually symlink construct libraries together or
use a monorepo tool: in those cases, multiple copies of the <code>constructs</code>
library can be accidentally installed, and <code>instanceof</code> will behave
unpredictably. It is safest to avoid using <code>instanceof</code>, and using
this type-testing method instead.</p>
</div></div><h4 class="tsd-parameters-title">Parameters</h4><ul class="tsd-parameters"><li><h5>x: <span class="tsd-signature-type">any</span></h5><div class="tsd-comment tsd-typography"><div class="lead">
<p>Any object</p>
</div></div></li></ul><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">x</span><span class="tsd-signature-symbol"> is </span><span class="tsd-signature-type">Construct</span></h4><div><p>true if <code>x</code> is an object created from a class which extends <code>Construct</code>.</p>
</div></li></ul></section></section></div><div class="col-4 col-menu menu-sticky-wrap menu-highlight"><nav class="tsd-navigation primary"><ul><li class=""><a href="../modules.html">Exports</a></li></ul></nav><nav class="tsd-navigation secondary menu-sticky"><ul><li class="current tsd-kind-class"><a href="ApplicationTargetGroup.html" class="tsd-kind-icon">Application<wbr/>Target<wbr/>Group</a><ul><li class="tsd-kind-constructor tsd-parent-kind-class tsd-is-overwrite"><a href="ApplicationTargetGroup.html#constructor" class="tsd-kind-icon">constructor</a></li><li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a href="ApplicationTargetGroup.html#node" class="tsd-kind-icon">node</a></li><li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a href="ApplicationTargetGroup.html#stack" class="tsd-kind-icon">stack</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="ApplicationTargetGroup.html#targetGroup" class="tsd-kind-icon">target<wbr/>Group</a></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited tsd-is-external"><a href="ApplicationTargetGroup.html#toString" class="tsd-kind-icon">to<wbr/>String</a></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited tsd-is-static tsd-is-external"><a href="ApplicationTargetGroup.html#isConstruct" class="tsd-kind-icon">is<wbr/>Construct</a></li></ul></li></ul></nav></div></div></div><footer class="with-border-bottom"><div class="container"><h2>Legend</h2><div class="tsd-legend-group"><ul class="tsd-legend"><li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li><li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li><li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited property</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected property</span></li></ul></div><h2>Settings</h2><p>Theme <select id="theme"><option value="os">OS</option><option value="light">Light</option><option value="dark">Dark</option></select></p></div></footer><div class="container tsd-generator"><p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p></div><div class="overlay"></div><script src="../assets/main.js"></script></body></html>