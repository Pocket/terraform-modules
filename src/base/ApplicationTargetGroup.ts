import { AlbTargetGroup } from '@cdktf/provider-aws/lib/alb-target-group';
import { TerraformMetaArguments } from 'cdktf';
import { Construct } from 'constructs';

export interface ApplicationTargetGroupProps extends TerraformMetaArguments {
  shortName: string;
  vpcId: string;
  healthCheckPath: string;
  tags?: { [key: string]: string };
}

export class ApplicationTargetGroup extends Construct {
  public readonly targetGroup: AlbTargetGroup;

  constructor(
    scope: Construct,
    name: string,
    config: ApplicationTargetGroupProps,
  ) {
    super(scope, name);

    this.targetGroup = new AlbTargetGroup(this, 'ecs_target_group', {
      namePrefix: config.shortName,
      protocol: 'HTTP',
      vpcId: config.vpcId,
      tags: config.tags,
      targetType: 'ip',
      port: 80,
      deregistrationDelay: '120',
      healthCheck: {
        interval: 15,
        path: config.healthCheckPath,
        protocol: 'HTTP',
        healthyThreshold: 5,
        unhealthyThreshold: 3,
      },
      provider: config.provider,
    });
  }
}
