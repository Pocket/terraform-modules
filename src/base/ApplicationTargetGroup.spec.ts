import { Testing, TerraformStack } from 'cdktf';
import { ApplicationTargetGroup } from './ApplicationTargetGroup';

test('renders a Target Group without tags', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new ApplicationTargetGroup(stack, 'testTargetGroup', {
    shortName: 'ABC',
    vpcId: '123',
    healthCheckPath: '/',
  });

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders a Target Group with tags', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new ApplicationTargetGroup(stack, 'testTargetGroup', {
    shortName: 'A1BC',
    vpcId: '123',
    healthCheckPath: '/',
    tags: {
      name: 'thedude',
      hobby: 'bowling',
    },
  });

  expect(Testing.synth(stack)).toMatchSnapshot();
});
