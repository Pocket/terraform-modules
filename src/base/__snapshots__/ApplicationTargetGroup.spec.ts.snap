// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Target Group with tags 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_alb_target_group\\": {
      \\"testTargetGroup_ecs_target_group_A0C76C7F\\": {
        \\"deregistration_delay\\": 120,
        \\"name_prefix\\": \\"A1BC\\",
        \\"port\\": 80,
        \\"protocol\\": \\"HTTP\\",
        \\"tags\\": {
          \\"name\\": \\"thedude\\",
          \\"hobby\\": \\"bowling\\"
        },
        \\"target_type\\": \\"ip\\",
        \\"vpc_id\\": \\"123\\",
        \\"health_check\\": [
          {
            \\"interval\\": 15,
            \\"path\\": \\"/\\",
            \\"protocol\\": \\"HTTP\\",
            \\"healthyThreshold\\": 5,
            \\"unhealthyThreshold\\": 3
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testTargetGroup/ecs_target_group\\",
            \\"uniqueId\\": \\"testTargetGroup_ecs_target_group_A0C76C7F\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders a Target Group without tags 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_alb_target_group\\": {
      \\"testTargetGroup_ecs_target_group_A0C76C7F\\": {
        \\"deregistration_delay\\": 120,
        \\"name_prefix\\": \\"ABC\\",
        \\"port\\": 80,
        \\"protocol\\": \\"HTTP\\",
        \\"target_type\\": \\"ip\\",
        \\"vpc_id\\": \\"123\\",
        \\"health_check\\": [
          {
            \\"interval\\": 15,
            \\"path\\": \\"/\\",
            \\"protocol\\": \\"HTTP\\",
            \\"healthyThreshold\\": 5,
            \\"unhealthyThreshold\\": 3
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testTargetGroup/ecs_target_group\\",
            \\"uniqueId\\": \\"testTargetGroup_ecs_target_group_A0C76C7F\\"
          }
        }
      }
    }
  }
}"
`;
